readme.md

```diff
@@ -101,3 +101,176 @@ In the console...
+
+- __Creating__ new artists
+
+  - Trying to follow the links to the new artist page will lead to the same "Sinatra doesn't know this ditty" error.
+
+  - In the artist controller, define a route for GET "artists/new" and create your new page: views/artists/new.erb. The new page should have a [form which makes a POST request](https://github.com/ga-dc/curriculum/blob/master/04-ruby-mvc-sinatra/sinatra-rest/rest.md#post) to "/artists"
+
+  - Now in the artist controller when you define the route for POST "/artists", use the .create method of your Artist Model (like in [Part 2.3](##part-2-3) above) to create a record of an artist in the database using params to access the information your user posted in the form. After we create our artist, redirect to the artists index where you should see your new artist added to the list.
+
+- __Showing__ an artist
+
+  - Thinking back to our artist index page, a plain text list of the artists in the database isn't super useful, even with the very impressive ability to add to that list. What we would like to do is give each individual artist a page. ERB is going to be an enormous asset here. You can define a generic template for showing an artist and then provide the view an artist instance with which to render the form.
+
+    - The challenge here is to [take information from](https://github.com/ga-dc/curriculum/tree/master/04-ruby-mvc-sinatra/sinatra-rest#the-params-hash) the GET artists/:id and use that with ActiveRecord to retrieve the artist instance requested. Spend a little time in pry playing with these ideas.
+
+    - HTML forms only support GET and POST methods so we need to use a work-around to use our PUT and DELETE methods we need for updating and deleting artists. We create a form with `action='/artist/<id here>' method='put'` and any form elements we need but the first child of the form should be `<input type="hidden" name="_method" value="put">` or `value="delete"` for the delete route
+
+    - The form for deleting doesn't need its own page so can go on the Artist page however we should link to a new page which holds the edit form.
+
+- Finishing the routes
+
+  - Using the techniques covered so far, you have the means implement the 7 RESTful routes discussed above for artists  and create the views for your routes to respond with (giving your user full CRUD functionality over your database!).
+
+### Part 3.3 (Bonus)
+
+- If you are comfortable, begin implementing routes for songs -- consider carefully how a user wants to access songs. Remember that CRUD and ISNECUD are more ideas than hard and fast rules and we structure our app in a way that makes sense to how it will be used.
+
+### Part 3.4 (Super bonus)
+
+- Add a whole new model of playlists to the app
+
+### Part 3.5 (You're either time-traveling or not sleeping bonus)
+
+- Add users and account management
```

Comment:
#### Where to go next

Check out [HAML](http://haml.info/)! Convert your ERB views to use HAML